src/Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM alpine:3.18
+
+COPY requirements.txt .
+COPY main.py .
+
+RUN apk update && apk add --update --no-cache \
+	python3 \
+	py3-pip && \
+	ln -sf python3 /usr/bin/python # Makes 'python' available in path
+
+RUN pip install -r /requirements.txt --no-cache-dir && \
+    rm /requirements.txt
+
+CMD ["python", "/main.py"]
```

Comment:
This would be `ENTRYPOINT` rather than `CMD` because commands get overwritten by user input.

With `CMD ["python", "/main.py"]` then `docker run canary --foo=bar` will execute `--foo=bar` which is a terminal error.

With `ENTRYPOINT ["python", "/main.py"]` you get `python /main.py --foo=bar`.

For reference you can combine them, the following would force the help message to be shown unless user input was given.
```
ENTRYPOINT ["python", "/main.py"]
CMD ["--help"]
```

Reply:
Would we want it to automatically show the help message?

Reply:
No, just to highlight the usage.